hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/BurnPrecompile.java

```diff
@@ -145,6 +145,10 @@ public static BurnWrapper getBurnWrapper(final Bytes input, @NonNull final Syste
         final var fungibleAmount = SystemContractAbis.toLongSafely(decodedArguments.get(1));
         final var serialNumbers = (long[]) decodedArguments.get(2);
 
+        if (fungibleAmount < 0 || (fungibleAmount == 0 && serialNumbers.length == 0)) {
+            throw new IllegalArgumentException("Illegal amount of tokens to burn");
+        }
+
         if (fungibleAmount > 0) {
```

Comment:
Shouldn't this be >=0 ?

Reply:
My understanding is that since we did not break up this call into two separate calls (1 for FT and 1 for NFT), amount of values <= 0 is used to signal that the burn is for an NFT and > 0 for FT.  

The real answer is that we should look up the token type from the token id and then see if the passed in amount makes sense but because of time sensitivity, it was decided to do this for now and further improvements at a later time.

Reply:
Makes sense. Thanks @lukelee-sl